File: src/types/ChatModelVersion.ts
// Interfaces for ChatModelVersion update and retrieval processes

/**
 * Interface for the request to read a ChatModel's ChatModelVersions.
 */
export interface ReadChatModelVersionsRequest {
  chat_model_uuid: string;
}

/**
 * Interface for the request to read a ChatModelVersion's information.
 */
export interface ReadChatModelVersionRequest {
  uuid: string;
}

/**
 * Interface for the request to update the published ChatModelVersion.
 */
export interface UpdatePublishedChatModelVersionRequest {
  uuid: string; // ChatModelVersion UUID
  project_uuid: string;
  project_version: string;
  previous_published_version_uuid: string;
}

/**
 * Interface for the request to update a ChatModelVersion's tags.
 */
export interface UpdateChatModelVersionTagsRequest {
  uuid: string;
  tags: Array<string>;
}

/**
 * Interface for the request to update a ChatModelVersion's memo.
 */
export interface UpdateChatModelVersionMemoRequest {
  uuid: string;
  memo: string;
}

/**
 * General interface for representing a ChatModelVersion in the system.
 */
export interface ChatModelVersion {
  id: number;
  created_at: string;
  uuid: string;
  version: number;
  model: string;
  system_prompt: string;
  is_published: boolean;
  functions: Array<string>;
  is_ab_test: boolean;
  ratio: number;
  from_version: number;
  tags: Array<string>;
  memo: string;
  chat_model_uuid: string;
}

File: backend/api/web/models/chat_model_version.py
"""APIs for ChatModelVersion"""
from typing import List, Optional
from modules.types import PMObject


class ChatModelVersionInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    version: int
    model: str
    system_prompt: str
    is_published: bool

    functions: Optional[List[str]] = None
    is_ab_test: Optional[bool] = None
    ratio: Optional[float] = None
    from_version: Optional[int] = None
    tags: Optional[List[str]] = None
    memo: Optional[str] = None

    chat_model_uuid: str


class UpdatePublishedChatModelVersionBody(PMObject):
    project_uuid: str
    project_version: int
    previous_published_version_uuid: Optional[str] = None


class UpdateChatModelVersionTagsBody(PMObject):
    tags: Optional[List[str]] = None
